pkg/tests/schema_pulumi_test.go

```diff
@@ -4159,3 +4012,3298 @@ func TestMakeTerraformResultNilVsEmptyMap(t *testing.T) {
+	t.Setenv("PULUMI_TF_BRIDGE_ACCURATE_BRIDGE_PREVIEW", "true")
+	resMap := map[string]*schema.Resource{
+		"prov_test": {
+			Schema: map[string]*schema.Schema{
+				"test": {
+					Type:     schema.TypeSet,
+					Optional: true,
+					Elem: &schema.Schema{
+						Type: schema.TypeString,
+					},
+				},
+			},
+		},
+		"prov_aux": {
+			Schema: map[string]*schema.Schema{
+				"aux": {
+					Type:     schema.TypeString,
+					Computed: true,
+					Optional: true,
+				},
+			},
+			CreateContext: func(_ context.Context, d *schema.ResourceData, _ interface{}) diag.Diagnostics {
+				d.SetId("aux")
+				err := d.Set("aux", "aux")
+				require.NoError(t, err)
+				return nil
+			},
+		},
+	}
+
+	t.Run("empty to unknown element", func(t *testing.T) {
+		testDetailedDiffWithUnknowns(t, resMap, "${auxRes.aux}",
+			[]interface{}{},
+			[]interface{}{"UNKNOWN"},
+			autogold.Expect(`Previewing update (test):
+  pulumi:pulumi:Stack: (same)
+    [urn=urn:pulumi:test::test::pulumi:pulumi:Stack::test-test]
+    + prov:index/aux:Aux: (create)
+        [urn=urn:pulumi:test::test::prov:index/aux:Aux::auxRes]
+    ~ prov:index/test:Test: (update)
+        [id=newid]
+        [urn=urn:pulumi:test::test::prov:index/test:Test::mainRes]
+      + tests: [
+      +     [0]: output<string>
+        ]
+    --outputs:--
+  + testOut: output<string>
+Resources:
+    + 1 to create
+    ~ 1 to update
+    2 changes. 1 unchanged
+`),
+			autogold.Expect(map[string]interface{}{"tests": map[string]interface{}{}}))
+	})
+
+	t.Run("non-empty to unknown element", func(t *testing.T) {
+		testDetailedDiffWithUnknowns(t, resMap, "${auxRes.aux}",
+			[]interface{}{"val1"},
+			[]interface{}{"UNKNOWN"},
+			autogold.Expect(`Previewing update (test):
+  pulumi:pulumi:Stack: (same)
+    [urn=urn:pulumi:test::test::pulumi:pulumi:Stack::test-test]
+    + prov:index/aux:Aux: (create)
+        [urn=urn:pulumi:test::test::prov:index/aux:Aux::auxRes]
+    ~ prov:index/test:Test: (update)
+        [id=newid]
+        [urn=urn:pulumi:test::test::prov:index/test:Test::mainRes]
+      ~ tests: [
+          ~ [0]: "val1" => output<string>
+        ]
+Resources:
+    + 1 to create
+    ~ 1 to update
+    2 changes. 1 unchanged
+`),
+			autogold.Expect(map[string]interface{}{"tests": map[string]interface{}{"kind": "UPDATE"}}))
+	})
+
+	t.Run("unknown element added front", func(t *testing.T) {
+		testDetailedDiffWithUnknowns(t, resMap, "${auxRes.aux}",
+			[]interface{}{"val2", "val3"},
+			[]interface{}{"UNKNOWN", "val2", "val3"},
+			autogold.Expect(`Previewing update (test):
+  pulumi:pulumi:Stack: (same)
+    [urn=urn:pulumi:test::test::pulumi:pulumi:Stack::test-test]
+    + prov:index/aux:Aux: (create)
+        [urn=urn:pulumi:test::test::prov:index/aux:Aux::auxRes]
+    ~ prov:index/test:Test: (update)
+        [id=newid]
+        [urn=urn:pulumi:test::test::prov:index/test:Test::mainRes]
+      ~ tests: [
+          ~ [0]: "val2" => output<string>
```

Comment:
This isn't right anymore. We don't know how many elements the set will have. I think we might want to bail and return `~tests: output` to mark the entire collection as unknown.

Reply:
This is generated by the CLI. We have no way of telling it to bail